### PR TITLE
Flush only if data available

### PR DESCRIPTION
**Pull request by dominikwerder**
### Issue

- Closes DM-1261

### Description of work

- Only attempt to flush if data is available
- Was not an actual bug, but triggered code analysis lint
- ... and with this update we don't do unnecessary work, too.

### CODE REVIEW

File: src/h5.cpp [hunk 1]
```diff
@@ -436,6 +436,9 @@ append_ret h5d_chunked_1d<T>::append_data_1d(T const *data, hsize_t nlen) {
 }
 
 template <typename T> AppendResult h5d_chunked_1d<T>::flush_buf() {
+  if (buf_n == 0) {
+    return AppendResult::OK;
+  }
```
Comment (matthew-d-jones) [inline src/h5.cpp]: I'd rather avoid having more exit points from the function by wrapping the existing code in `if (buf_n > 0)` instead.

File: Jenkinsfile [hunk 2]
```diff
@@ -231,7 +231,7 @@ node {
     }
   }
 
-  builders['macOS'] = get_macos_pipeline()
+  //builders['macOS'] = get_macos_pipeline()
```
Comment (matthew-d-jones) [inline Jenkinsfile]: I think it is best to keep this enabled. The mac build usually passes on at least one of the two builds which Jenkins does, and we can go ahead and merge in this case.

File: src/h5.cpp (hunk 1 above)
Comment (dominikwerder) [reply src/h5.cpp]: The LLVM coding style favors early exits.

If a function has multiple exit points in nested logic, then I'd also vote to reduce the exit points.

But early-exit on pre-condition test is IMHO fine.

Comment (matthew-d-jones) [reply src/h5.cpp]: "early exit" means don't execute any other code before exiting.
I'm proposing:
```cpp
template <typename T> AppendResult h5d_chunked_2d<T>::flush_buf() {
  if (buf_n > 0) {
    auto wr = DataSet.append_data_2d(reinterpret_cast<T *>(buf.data()),
                                     buf_n / sizeof(T));
    if (wr.status != AppendResult::OK) {
      return wr.status;
    }
    buf_n = 0;
  }
  return AppendResult::OK;
}
```

Comment (dominikwerder) [reply src/h5.cpp]: > "early exit" means don't execute any other code before exiting.

I don't understand this comment. Isn't that exactly the case in my proposed version?

```cpp
template <typename T> AppendResult h5d_chunked_1d<T>::flush_buf() {
  if (buf_n == 0) {
    return AppendResult::OK;
  }
```

Nothing gets done before that return.

Comment (dominikwerder) [reply src/h5.cpp]: For reference:

[LLVM style guide on early exit](https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code)

**Comment by dominikwerder** [comment]
Both `branch` and `PR` tests have passed, only OSX hangs in both cases.

File: src/h5.cpp (hunk 1 above)
Comment (matthew-d-jones) [reply src/h5.cpp]: LLVM does say "if the body of the 'if' is large". But it isn't worth further disagreement so let's get this merged.
I didn't mean to imply that your code didn't exit early, just that my suggestion exits just as early and avoids multiple exit points, which is why I prefer it.

**Comment by matthew-d-jones** [comment]
It's still executing on OSX, I'll merge as soon as it has passed on one of the jobs.

**Pull request merged**